FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ */
+
+/*
+ * Demo for showing use of the managed MQTT API shared between multiple tasks.
+ * This demo uses a thread safe queue to hold commands for interacting with the
+ * MQTT API. A command task processes commands from the queue while other tasks
+ * enqueue them. In addition to the command task, this demo uses one task for
+ * publishing messages to the MQTT broker and another for receiving them via
+ * an MQTT subscription. Each task has a queue to hold received publish messages,
+ * and the command task pushes incoming publishes to the queue of each task
+ * that is subscribed to the incoming topic.
+ *
+ * !!! NOTE !!!
+ * This MQTT demo does not authenticate the server nor the client.
+ * Hence, this demo should not be used as production ready code.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* MQTT library includes. */
+#include "mqtt.h"
+
+/* Transport interface include. */
+#include "plaintext_freertos.h"
+
+/**
+ * These configuration settings are required to run the demo.
+ * Throw compilation error if the below configs are not defined.
+ */
+#ifndef CLIENT_IDENTIFIER
+    #error "Please define a unique CLIENT_IDENTIFIER."
+#endif
+#ifndef BROKER_ENDPOINT
+    #error "Please define BROKER_ENDPOINT."
+#endif
+
+/**
+ * Provide default values for undefined configuration settings.
+ */
+#ifndef BROKER_PORT
+    #define BROKER_PORT    ( 1883 )
+#endif
+
+#ifndef NETWORK_BUFFER_SIZE
+    #define NETWORK_BUFFER_SIZE    ( 1024U )
+#endif
+
+/**
+ * @brief Length of client identifier.
+ */
+#define CLIENT_IDENTIFIER_LENGTH            ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) - 1 ) )
```

Comment:
Please take a look at the naming conventions for the MACROS for FreeRTOS https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html